packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -39,9 +40,44 @@ import { __ } from '@wordpress/i18n';
 import BlockInspectorButton from './block-inspector-button';
 import { store as editPostStore } from '../../store';
 
-export default function VisualEditor( { styles } ) {
-	const ref = useRef();
+function MaybeIframe( {
+	children,
+	contentRef,
+	isTemplateMode,
+	styles,
+	style,
+} ) {
+	const ref = useMouseMoveTypingReset();
```

Comment:
What is this hook about?

Reply:
It's the same one we have in the site editor. It tracks mouse movement outside the iframe for setting isTyping to false. With #31280, it would be integrated in the block editor so the implementor doesn't need to think about it.